### PR TITLE
Fix #1106.

### LINKED ISSUE #1106: Path dependant types are lost in denotation signatures
```scala
object a {
  class Path {
    class B { class C }
    implicit val x = new B
    implicit val y = new x.C
  }
  implicit val b = new Path().x
}
// obtained
_empty_.a.Path#y. => implicit val y: C
  [0..1): C => _empty_.a.Path#B#C#
// expected
_empty_.a.Path#y. => implicit val y: Path.this.x.C
  [10..11): x => _empty_.a.Path#x.
  [12..13): C => _empty_.a.Path#B#C#
```

### PR DESCRIPTION
**Pull request by olafurpg**
I'm not sure if this solution is acceptable, but it seems to handle the problematic case I could come up with.

### CODE REVIEW

File: scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/PrinterOps.scala [hunk 1]
```diff
@@ -320,6 +320,13 @@ trait PrinterOps { self: DatabaseOps =>
     }
 
     // + scalac deviation
+    object PathDependentType {
+      def unapply(arg: g.Type): Option[g.Symbol] = arg match {
+        case SingleType(ThisType(_), sym) if sym.isTerm && !sym.isModule =>
```
Comment (xeno-by) [inline scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/PrinterOps.scala]: `SingleType(_, sym)`

File: scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/PrinterOps.scala [hunk 2]
```diff
@@ -320,6 +320,13 @@ trait PrinterOps { self: DatabaseOps =>
     }
 
     // + scalac deviation
+    object PathDependentType {
```
Comment (xeno-by) [inline scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/PrinterOps.scala]: `PathDependentPrefix`

Comment (olafurpg) [reply scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/PrinterOps.scala]: Done.

File: scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/PrinterOps.scala (hunk 1 above)
Comment (olafurpg) [reply scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/PrinterOps.scala]: Done.

**Pull request merged**